### PR TITLE
Add TMS (Tile Map Service) support

### PR DESCRIPTION
**Pull request by ronitjadhav**
## Overview
This PR adds support for TMS (Tile Map Service) to the ogc-client library, extending its capabilities beyond WMS, WMTS, WFS, and OGC-API to now include TMS-compliant services according to the [Tile Map Service Specification](https://wiki.osgeo.org/wiki/Tile_Map_Service_Specification).

### TMS Endpoint
Implemented `TmsEndpoint` class with the following methods:

* **getEndpointInfo()**: Retrieves basic information about the TMS endpoint, including title, abstract, and keywords
* **getAllTileMaps()**: Fetches available tile maps from the TMS service, filtering to a specific tile map if the endpoint points to one
* **getTileMapInfo(href)**: Retrieves detailed information about a specific tile map by URL

The class handles proper caching of service data through internal methods:
* Private `getServiceData()`: Fetches and caches TMS service data
* Private `getFilteredData()`: Applies filtering based on the endpoint URL

### Utility Functions
* **normalizeUrl(url)**: Ensures consistent URL formatting with lowercase hostname and trailing slash
* **getParentPath(url)**: Extracts parent path from versioned TMS URLs
* **fetchXml(url)**: Fetches XML content with robust error handling for:
  * Network failures
  * Non-XML responses
  * XML with or without declaration

## Implementation Details
The implementation closely follows the TMS specification while maintaining consistency with the API design pattern used throughout ogc-client. This ensures a familiar interface for developers already using the library for other OGC services.

## Current Limitations
* Root resource handling as described in the TMS specification is not yet implemented and will be addressed in a future update

## Testing
Comprehensive unit tests cover:
* URL normalization with various edge cases
* XML fetching with different response types
* Error handling for network issues and invalid content
* TmsEndpoint class functionality for retrieving endpoint information and tile maps

## Usage Example
```typescript
import { TmsEndpoint } from '@camptocamp/ogc-client';

// Create a TMS endpoint
const endpoint = new TmsEndpoint('https://ahocevar.com/geoserver/gwc/service/tms/1.0.0/');

// Get endpoint information
const info = await endpoint.tileMapServiceInfo;
console.log(info.title, info.abstract);

// Get all available tile maps
const tileMaps = await endpoint.allTileMaps;

// Get detailed information for a specific tile map
const tileMapDetails = await endpoint.getTileMapInfo(tileMaps[0].href);

### CODE REVIEW

File: app/src/components/tms/TileMapDetails.vue [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  computed: {
+    boundingBoxValues() {
+      return Array.isArray(this.tileMap.boundingBox)
+        ? this.tileMap.boundingBox
+        : [
+            this.tileMap.boundingBox?.minx,
+            this.tileMap.boundingBox?.miny,
+            this.tileMap.boundingBox?.maxx,
+            this.tileMap.boundingBox?.maxy,
+          ];
```
Comment (jahow) [inline app/src/components/tms/TileMapDetails.vue:103]: Is that something that application code would be expected to do? IT sounds like the Bbox should always be an array
